### PR TITLE
configury: fix XRC detection on OFED < 3.12

### PR DESCRIPTION
**Pull request by ggouaillardet**
Thanks Paul Hargrove for the report

(back-ported from commit open-mpi/ompi@9a49d42f81e3ec2b63aa4e7805d4c5e44a90abab)

### CODE REVIEW

**Comment by ggouaillardet** [comment]
:bot:assign: @miked-mellanox 
:bot:label:bug
:bot:milestone:v1.10.0

**Comment by ggouaillardet** [comment]
@miked-mellanox could you please review this ?

**Comment by jsquyres** [comment]
:-1: 

This commit on master is bad -- it is causing builds to break on RHEL 6.5:

```
make[2]: Entering directory `/home/jsquyres/git/ompi-bisect/opal/mca/btl/openib'
  CC       btl_openib.lo
In file included from btl_openib_proc.h:27:0,
                 from btl_openib.c:52:
btl_openib_endpoint.h: In function 'post_send':
btl_openib_endpoint.h:616:16: error: 'struct ibv_send_wr' has no member named 'xrc_remote_srq_num'
         sr_desc->xrc_remote_srq_num = ep->rem_info.rem_srqs[qp].rem_srq_num;
                ^
btl_openib.c: In function 'create_srq':
btl_openib.c:354:21: error: implicit declaration of function 'ibv_create_xrc_sr' [-Werror=implicit-function-declaration]
                     ibv_create_xrc_srq(openib_btl->device->ib_pd,
                     ^
btl_openib.c:353:50: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
                 openib_btl->qps[qp].u.srq_qp.srq =
                                                  ^
btl_openib.c: In function 'prepare_device_for_use':
btl_openib.c:678:50: error: 'IBV_DEVICE_XRC' undeclared (first use in this function)
     if (!(device->ib_dev_attr.device_cap_flags & IBV_DEVICE_XRC) &&
                                                  ^
btl_openib.c:678:50: note: each undeclared identifier is reported only once for each function it appears in
cc1: some warnings being treated as errors
```

**Comment by jsquyres** [comment]
See full details of the failed build on this gist: https://gist.github.com/ddeb601839338fe7f152

File: config/ompi_check_openfabrics.m4 [hunk 1]
```diff
@@ -168,15 +168,19 @@ AC_DEFUN([OMPI_CHECK_OPENFABRICS],[
            # ibv_cmd_open_xrcd (aka XRC Domains) was added in  OFED 3.12
            if test "$enable_connectx_xrc" = "yes"; then
                $1_have_xrc=1
-               AC_CHECK_FUNCS([ibv_create_xrc_rcv_qp ibv_cmd_open_xrcd],
+               AC_CHECK_FUNCS([ibv_create_xrc_rcv_qp],
                               [], [$1_have_xrc=0])
-               AC_CHECK_DECLS([IBV_SRQT_XRC],
```
Comment (ggouaillardet) [inline config/ompi_check_openfabrics.m4:173]: IBV_SRQT_XRC is specific to XRC domains, that came with OFED 3.12, so the this test disables XRC on OFED < 3.12 :-(
Instead, we can test IBV_DEVICE_XRC that is common to all OFEDs, but is not part of the RHEL6 broken headers

**Comment by ggouaillardet** [comment]
:bot:nolabel:pushed-back
this PR has been updated based on open-mpi/ompi#718
it should now correctly handle broken ibverbs devel headers

**Pull request merged**